### PR TITLE
add docs for using crictl

### PR DESCRIPTION
**Pull request by mikebrow**
Addresses #96; adds documentation describing how to use crictl to aid in debugging CRI-Containerd.

Signed-off-by: Mike Brown <brownwm@us.ibm.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ CONTAINER_RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT='/var/run/cri-containerd.soc
 ```
 ### Test
 See [here](./docs/testing.md) for information about test.
+## Using the crictl Client
```
Comment (Random-Liu) [inline README.md]: Debug?
Not sure. Just an idea here. :)

File: docs/crictl.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
+CRICTL Client User Guide
+========================
+This document presumes you already have `containerd` and `cri-containerd`
+installed and running.
+
+This document is for developers who wish to debug/inspect and manage their pods,
```
Comment (Random-Liu) [inline docs/crictl.md]: debug/inspect -> debug, inspect?

File: README.md [hunk 3]
```diff
@@ -105,6 +105,9 @@ CONTAINER_RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT='/var/run/cri-containerd.soc
...
+See [here](./docs/crictl.md) for information about using `crictl` to debug
+pods, containers, and images.
```
Comment (abhi) [inline README.md]: /s/, and/ and/

File: docs/crictl.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+containers, and container images.
```
Comment (abhi) [inline docs/crictl.md]: oh is it a thing to , and ?

File: docs/crictl.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+Before generating issues against this document, `containerd`, `cri-containerd`,
+or `crictl` please make sure the issue has not already been submitted.
+
+## Install crictl
+If you have not already installed crictl please install the version compatible
+with the `cri-containerd` you are using. The current version is specified
+[here](./hack/version). A helper command has been included to install the
+dependencies at the right version for you:
+```shell
+# make install.deps
```
Comment (abhi) [inline docs/crictl.md]: may be a note ? That install.deps does do a whole bunch of things other than crictl installation ?

File: docs/crictl.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+  "status": {
+    "conditions": [
+      {
+        "type": "RuntimeReady",
+        "status": true,
+        "reason": "",
+        "message": ""
+      },
+      {
+        "type": "NetworkReady",
```
Comment (abhi) [inline docs/crictl.md]: Just for my knowledge , Is this true if CNI plugin is installed and running ?

File: README.md (hunk 3 above)
Comment (mikebrow) [reply README.md]: The comma is optional, but is correct as well.

File: docs/crictl.md (hunk 4 above)
Comment (mikebrow) [reply docs/crictl.md]: :-) yeah

File: docs/crictl.md (hunk 5 above)
Comment (mikebrow) [reply docs/crictl.md]: thought about it.. they'll figure that out :-)

Comment (mikebrow) [reply docs/crictl.md]: and we might want to split em up..

File: docs/crictl.md (hunk 6 above)
Comment (mikebrow) [reply docs/crictl.md]: yes

Comment (Random-Liu) [reply docs/crictl.md]: Yeah, if cni plugin and configuration are installed.

File: docs/crictl.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
+CRICTL Client User Guide
```
Comment (Random-Liu) [inline docs/crictl.md]: CRICTL User Guide

File: docs/crictl.md (hunk 5 above)
Comment (Random-Liu) [reply docs/crictl.md]: Actually we package `crictl` into our release tarball, can we also mention that?
If you are a user, your deployement should have install `crictl` for you. If not, get it from release tarball.
If you are a developer, use `make install.deps`.

We should also mention the configuration file `/etc/crictl.yaml`. We also have it in the tarball.

Comment (Random-Liu) [inline docs/crictl.md]: Can we use `console` + `$` instead of `shell` + `#` here?
`#` is treated as a comment in shell.

File: docs/crictl.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        "attempt": 1,
+        "uid": "hdishd83djaidwnduwk28bcsb"
+    },
+    "linux": {
+    }
+}
+
+# crictl runs sandbox-config.json
+e1c83b0b8d481d4af8ba98d5f7812577fc175a37b10dc824335951f52addbb4e
+# crictl sandboxes
```
Comment (Random-Liu) [inline docs/crictl.md]: Can you generate the result with newest `crictl`, this is old.

File: docs/crictl.md (hunk 5 above)
Comment (abhi) [reply docs/crictl.md]: sgtm

File: docs/crictl.md (hunk 4 above)
Comment (abhi) [reply docs/crictl.md]: fine fine

File: docs/crictl.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+  "command": [
+      "top"
+  ],
+  "linux": {
+  }
+}
+
+# crictl create e1c83 container-config.json sandbox-config.json
+0a2c761303163f2acaaeaee07d2ba143ee4cea7e3bde3d32190e2a36525c8a05
+# crictl ps
```
Comment (Random-Liu) [inline docs/crictl.md]: Also the `ps` result is old.

File: docs/crictl.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+```shell
+crictl version
+Version:  0.1.0
+RuntimeName:  cri-containerd
+RuntimeVersion:  1.0.0-alpha.1-167-g737efe7-dirty
+RuntimeApiVersion:  0.0.0
+```
+## Display Status & Configuration Information about Containerd & CRI-Containerd
+```shell
+crictl info
```
Comment (Random-Liu) [inline docs/crictl.md]: `$`

File: docs/crictl.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+```shell
+# crictl stats
+CONTAINER           CPU %               MEM                 DISK                INODES
+0a2c761303163f      0.00                991.2kB             16.38kB             6
+```
+* Other commands to manage the container include `stop ID` to stop a running
+container and `rm ID` to remove a container.
+## Display Version Information
+```shell
+crictl version
```
Comment (Random-Liu) [inline docs/crictl.md]: `$`

File: docs/crictl.md (hunk 7 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 2 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 5 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 8 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 9 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 10 above)
Comment (mikebrow) [reply docs/crictl.md]: done

**Comment by mikebrow** [comment]
comments addressed

**Comment by mikebrow** [comment]
Thanks for reviewing @abhi @Random-Liu

File: docs/crictl.md [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+developers:
+```shell
+$ make install.deps
+```
+* Note: The file named `/etc/crictl.yaml` is used to configure crictl
+so you don't have to repeatedly specify the runtime sock used to connect crictl
+to the container runtime:
+```
+# cat /etc/crictl.yaml
+runtime-endpoint: /var/run/dockershim.sock
```
Comment (Random-Liu) [inline docs/crictl.md]: cri-containerd.sock?

File: docs/crictl.md [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+
+This document is for developers who wish to debug, inspect, and manage their pods,
+containers, and container images.
+
+Before generating issues against this document, `containerd`, `cri-containerd`,
+or `crictl` please make sure the issue has not already been submitted.
+
+## Install crictl
+If you have not already installed crictl please install the version compatible
+with the `cri-containerd` you are using. If you are a user, your deployement
```
Comment (Random-Liu) [inline docs/crictl.md]: * If you are a user
* If you are a developer

File: docs/crictl.md [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+command has been included to install the dependencies at the right version for
+developers:
+```shell
+$ make install.deps
+```
+* Note: The file named `/etc/crictl.yaml` is used to configure crictl
+so you don't have to repeatedly specify the runtime sock used to connect crictl
+to the container runtime:
+```
+# cat /etc/crictl.yaml
```
Comment (Random-Liu) [inline docs/crictl.md]: $

File: docs/crictl.md [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+The current version of crictl is specified [here](./hack/version). A helper
+command has been included to install the dependencies at the right version for
+developers:
+```shell
+$ make install.deps
+```
+* Note: The file named `/etc/crictl.yaml` is used to configure crictl
+so you don't have to repeatedly specify the runtime sock used to connect crictl
+to the container runtime:
+```
```
Comment (Random-Liu) [inline docs/crictl.md]: console
and `s/shell/console/g`.

File: docs/crictl.md [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+        "uid": "hdishd83djaidwnduwk28bcsb"
+    },
+    "linux": {
+    }
+}
+
+$ crictl runs sandbox-config.json
+e1c83b0b8d481d4af8ba98d5f7812577fc175a37b10dc824335951f52addbb4e
+$ crictl sandboxes
+SANDBOX ID          CREATED             STATE               NAME                                                              NAMESPACE                                                ATTEMPT
```
Comment (Random-Liu) [inline docs/crictl.md]: Check the preview, too many spaces between NAMESPACE and ATTEMPT

File: docs/crictl.md (hunk 10 above)
Comment (Random-Liu) [reply docs/crictl.md]: Not done? :)

File: docs/crictl.md (hunk 11 above)
Comment (Random-Liu) [reply docs/crictl.md]: Not done?

File: docs/crictl.md (hunk 13 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 14 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 15 above)
Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 12 above)
Comment (mikebrow) [reply docs/crictl.md]: whoops :-)

Comment (mikebrow) [reply docs/crictl.md]: done

File: docs/crictl.md (hunk 16 above)
Comment (mikebrow) [reply docs/crictl.md]: done

**Pull request merged**